velox/functions/prestosql/Rand.h

```diff
@@ -56,4 +56,55 @@ struct RandFunction {
   }
 };
 
+template <typename T>
+struct SecureRandFunction {
+  static constexpr bool is_deterministic = false;
+
+  FOLLY_ALWAYS_INLINE void call(double& result) {
+    result = folly::Random::secureRandDouble01();
+  }
+
+  FOLLY_ALWAYS_INLINE void
+  call(double& out, const double lower, const double upper) {
+    VELOX_USER_CHECK_GT(
```

Comment:
Maybe have an common inline function (with template type for lower and upper) with just this check statement so that its not repeated in all the functions.